kbase-extension/static/kbase/js/widgets/narrative_core/upload/stagingAreaViewer.js

```diff
@@ -951,14 +951,17 @@ define([
             // get all of the selected checkbox file names and import type
             $(`input.${cssBaseClass}-body__checkbox-input:checked`).each(function () {
                 const importType = $(this).attr('data-type');
-                const importFile = $(this).attr('data-file-name');
+                let importFile = $(this).attr('data-file-name');
                 if (stagingAreaViewer.bulkImportTypes.includes(importType)) {
                     if (!(importType in bulkMapping)) {
                         bulkMapping[importType] = {
                             appId: stagingAreaViewer.uploaders.app_info[importType].app_id,
                             files: [],
                         };
                     }
+                    if (stagingAreaViewer.subpath) {
```

Comment:
This is the actual bugfix. Just make sure the subpath is appended to the file.

Though it may make sense to somehow, internally (as part of the file object? Or hidden in the row?) track that info. I think there's a feature/bug where selected files in subdirectories remain selected when navigating out of that subdirectory. Feature when it works as expected, possibly a bug in this case if the user's expecting files from both a subdirectory and root.